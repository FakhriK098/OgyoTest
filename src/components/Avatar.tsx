import { Image, StyleSheet, View } from 'react-native';
import React from 'react';
import { colors } from '@themes/colors';
import { IAvatarProps } from 'src/types/profile';

const Avatar = ({ url, width, height }: IAvatarProps) => {
  return (
    <View style={styles.containerAvatar}>
      <Image source={{ uri: url }} style={[styles.avatar, { width, height }]} />
    </View>
  );
};

export default Avatar;

const styles = StyleSheet.create({
  containerAvatar: {
    elevation: 8,
    shadowColor: colors.black,
    shadowOffset: {
      width: 0,
      height: 4,
    },
    shadowOpacity: 0.3,
    shadowRadius: 4.65,
    backgroundColor: colors.white,
    borderRadius: 50,
  },
  avatar: {
    borderRadius: 50,
  },
});
