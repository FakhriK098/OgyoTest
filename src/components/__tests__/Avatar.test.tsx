import React from 'react';
import { render } from '@testing-library/react-native';
import Avatar from '../Avatar';

describe('Avatar', () => {
  const defaultProps = {
    url: 'https://example.com/avatar.jpg',
    width: 100,
    height: 100,
  };

  it('should render correctly with provided props', () => {
    const { getByTestId } = render(<Avatar {...defaultProps} />);

    const image = getByTestId('avatar-image');
    expect(image).toBeTruthy();
    expect(image.props.source.uri).toBe('https://example.com/avatar.jpg');
    expect(image.props.style).toContainEqual({ width: 100, height: 100 });
  });

  it('should match snapshot', () => {
    const { toJSON } = render(<Avatar {...defaultProps} />);
    expect(toJSON()).toMatchSnapshot();
  });
});
