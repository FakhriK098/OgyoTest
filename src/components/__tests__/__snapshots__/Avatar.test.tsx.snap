// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar should match snapshot 1`] = `
<View
  style={
    {
      "backgroundColor": "#FFFFFF",
      "borderRadius": 50,
      "elevation": 8,
      "shadowColor": "#000000",
      "shadowOffset": {
        "height": 4,
        "width": 0,
      },
      "shadowOpacity": 0.3,
      "shadowRadius": 4.65,
    }
  }
>
  <Image
    source={
      {
        "uri": "https://example.com/avatar.jpg",
      }
    }
    style={
      [
        {
          "borderRadius": 50,
        },
        {
          "height": 100,
          "width": 100,
        },
      ]
    }
    testID="avatar-image"
  />
</View>
`;
